db/sql/sql_test.go

```diff
@@ -353,8 +360,189 @@ var _ = Describe("Sql", func() {
+			})
+			It("should create select query with one parameter", func() {
+				filter := map[string]interface{}{"test_string": "123"}
+
+				res, err := AddFilterToQuery(testSchema, query, filter, false)
+
+				Expect(err).ToNot(HaveOccurred())
+				resSql, param, err := res.ToSql()
+				Expect(err).ToNot(HaveOccurred())
+				expectedQuery = expectedQuery.Where(squirrel.Eq{"`test_string`": "123"})
+				expectedSql, expectedParam, _ := expectedQuery.ToSql()
+				Expect(resSql).To(Equal(expectedSql))
+				Expect(param).To(Equal(expectedParam))
+			})
+			It("should create conjunction for more then one parameter by default", func() {
+				filter := map[string]interface{}{"test_string": "123", "test_number": 42}
+
+				res, err := AddFilterToQuery(testSchema, query, filter, false)
+
+				Expect(err).ToNot(HaveOccurred())
+				resSql, param, err := res.ToSql()
+				Expect(err).ToNot(HaveOccurred())
+				expectedQuery = expectedQuery.Where(squirrel.Eq{"`test_string`": "123"}).Where(squirrel.Eq{"`test_number`": 42})
+				expectedSql, expectedParam, _ := expectedQuery.ToSql()
+				Expect(resSql).To(Equal(expectedSql))
+				Expect(param).To(Equal(expectedParam))
+			})
+		})
+		Context("Disjunction and conjunction", func() {
+			It("should create disjunction for more then one parameter", func() {
+				filter := map[string]interface{}{"__or__": []map[string]interface{}{
+					{
+						"property": "test_string",
+						"type":     "eq",
+						"value":    "123",
+					},
+					{
+						"property": "test_number",
+						"type":     "eq",
+						"value":    42,
+					},
+				}}
+
+				res, err := AddFilterToQuery(testSchema, query, filter, false)
+
+				Expect(err).ToNot(HaveOccurred())
+				resSql, param, err := res.ToSql()
+				Expect(err).ToNot(HaveOccurred())
+
+				expectedQuery = expectedQuery.Where(squirrel.Or{squirrel.Eq{"`test_string`": "123"}, squirrel.Eq{"`test_number`": 42}})
+				expectedSql, expectedParam, _ := expectedQuery.ToSql()
+				Expect(resSql).To(Equal(expectedSql))
+				Expect(param).To(Equal(expectedParam))
+			})
+			It("should create conjunction for more then one parameter", func() {
+				filter := map[string]interface{}{"__and__": []map[string]interface{}{
+					{
+						"property": "test_string",
+						"type":     "eq",
+						"value":    "123",
+					},
+					{
+						"property": "test_number",
+						"type":     "eq",
+						"value":    42,
+					},
+				}}
+
+				res, err := AddFilterToQuery(testSchema, query, filter, false)
+
+				Expect(err).ToNot(HaveOccurred())
+				resSql, param, err := res.ToSql()
+				Expect(err).ToNot(HaveOccurred())
+
+				expectedQuery = expectedQuery.Where(squirrel.And{squirrel.Eq{"`test_string`": "123"}, squirrel.Eq{"`test_number`": 42}})
+				expectedSql, expectedParam, _ := expectedQuery.ToSql()
+				Expect(resSql).To(Equal(expectedSql))
+				Expect(param).To(Equal(expectedParam))
+			})
+			It("should create disjunction of conjunctions for more then one parameter", func() {
```

Comment:
how does operator precedence work?
for example what does `filter := {"test_string": "123", "test_number": 42, "__or__": {{"property": ...}}` produce? is it
`(test_string = 123 AND test_number = 42) OR (...)` (which makes sense) or `(test_string = 123) AND (test_number = 42 OR (...))` (which does not to make sense due to Golangs map ordering)?
please add a test case for that

Reply:
none of this true.
in your example `filter := {"a": 1, "b": 2, "__or__": {{"property": "c", "value":3, "type":"eq"},{"property": "d", "value":4, "type":"neq"}}` it would produce `a = 1 AND b = 2 AND (c=3 OR d <> 4)`.
If you want `(a=1 and b=2) OR (c=3 OR d<> 4)` you just have to nest one operator in another i.e. 
```filter := {
"__or__":
  {"__and__":{
    {"property": "a", "value" 1, "type":"eq"},
    {"property": "b", "value" 2, "type":"eq"},
  },
  "__or__": {
    {"property": "c", "value":3, "type":"eq"},
    {"property": "d", "value":4, "type":"neq"}
  },
}
```